### PR TITLE
Issue #310 add tournament location

### LINKED ISSUE #310: Tournament location
As a tournament administrator I want to be able to set Location of the tournament so I can associate tournaments to locations.

AC:
- Add field for Location. Optional, up to 50 symbols, any letter, number, hyphens and apostrophes are allowed.

- [x] #317 Add location
- [x]  #318 Manual testing

### PR DESCRIPTION
**Pull request by NataliiaBidiuk**
### Summary
Add Location in the tournament, add in View and tests also
closes #310 

#### Design

- [ ] Web API layer contains as little logic as possible
- [ ] Domain objects use semantic names

#### Perfromance

- [ ] In case of any DB query was changed: attach generated SQL for review
- [ ] Number of DB roundtrips should be as low as possible

#### Unit Tests

- [ ] Naming: Initial state and expected result are properly stated
- [ ] Test follows [AAA](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) pattern.
- [ ] Test is Isolated
- [ ] Mock instances and expected instances do not share references
- [ ] When new property is added all related comparers are updated

### CODE REVIEW

File: src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Entities/ValidationConstants.cs [hunk 1]
```diff
@@ -13,14 +13,16 @@ internal static class ValidationConstants
...
+        {       
             public const short SCHEMA_STORAGE_OFFSET = 1900;
 
             public const int MAX_NAME_LENGTH = 60;
 
             public const int MAX_DESCRIPTION_LENGTH = 300;
 
             public const int MAX_URL_LENGTH = 255;
+
+            public static int MAX_LOCATION_LENGTH = 50;
```
Comment (volleymanagement-sonarqube) [inline src/VolleyManagement.Backend/VolleyManagement.Data.MsSql/Entities/ValidationConstants.cs]: CRITICAL Change the visibility of 'MAX_LOCATION_LENGTH' or make it 'const' or 'readonly'. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2223)

**Comment by volleymanagement-sonarqube** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 2 critical

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [TournamentValidationSpecification.cs#L15](https://github.com/VolleyManagement/volley-management/blob/bd52dde899cfd73e1783a5f117d544090efa3181/src/VolleyManagement.Backend/VolleyManagement.Domain/TournamentsAggregate/TournamentValidationSpecification.cs#L15): The Cyclomatic Complexity of this method is 11 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Comment by volleymanagement-sonarqube** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [TournamentValidationSpecification.cs#L15](https://github.com/VolleyManagement/volley-management/blob/eb727746fcf1ee9062d90070e40f240cd8bd9949/src/VolleyManagement.Backend/VolleyManagement.Domain/TournamentsAggregate/TournamentValidationSpecification.cs#L15): The Cyclomatic Complexity of this method is 11 which is greater than 10 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1541)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Comment by sdiachen** [comment]
You can start testing

File: src/VolleyManagement.Backend/VolleyManagement.Domain/TournamentsAggregate/TournamentValidationSpecification.cs [hunk 2]
```diff
@@ -14,37 +14,56 @@ public class TournamentValidationSpecification : ISpecification<Tournament>
...
+
+        private static bool IsSatisfiedByName(Tournament entity)
+        {
+            return (string.IsNullOrEmpty(entity.Name) || entity.Name.Length >= Constants.Tournament.MAX_NAME_LENGTH);
+        }
+
+        private static bool IsSatisfiedByDescription(Tournament entity)
+        {           
+            return (!string.IsNullOrEmpty(entity.Description)
+                    && entity.Description.Length >= Constants.Tournament.MAX_DESCRIPTION_LENGTH);
```
Comment (Myroslavv) [inline src/VolleyManagement.Backend/VolleyManagement.Domain/TournamentsAggregate/TournamentValidationSpecification.cs]: So if I pass null/empty string - it'll retun false. The same outcome awaits if I pass description with a valid length.

If you're checking if Description is invalid, try

`string.IsNullOrEmpty(entity.Description) || 
entity.Description.Length >= Constants.Tournament.MAX_DESCRIPTION_LENGTH);`

**Pull request merged**